### PR TITLE
Update boundary conditions interface

### PR DESCRIPTION
**Pull request by jkozdon**
Make the BC interfaces more general

### CODE REVIEW

File: src/Utilities/VariableTemplates/VariableTemplates.jl [hunk 1]
```diff
@@ -117,7 +117,7 @@ end
       offset += N
     elseif T <: StaticArray
       N = length(T)
-      retexpr = :(array[$(offset + 1):$(offset + N)] .= val)
+      retexpr = :(array[$(offset + 1):$(offset + N)] .= val[:])
```
Comment (charleskawczynski) [inline src/Utilities/VariableTemplates/VariableTemplates.jl]: Is this to fix shape mismatch with the symmetric matrix?

Comment (jkozdon) [reply src/Utilities/VariableTemplates/VariableTemplates.jl]: No, if you have a `SMatrix` in the `Vars`

Comment (jkozdon) [reply src/Utilities/VariableTemplates/VariableTemplates.jl]: Everywhere right now we have `SVector`

**Pull request merged**